rancher/dashboard
# Ensure that we show empty machine pools

Fixes #5773 

For RKE2, when you scale down a machine pool, there are no machine deployments in that pool, so we don't show the machine pool.

This PR fixes this - we ensure we have table groups for all of the machine pools - generated off of the pool info on the cluster resource.

SortableTable has been updated with a property that allows us to pass in the set of groups that always must be shown. We should use this mechanism for generally in other places to avoid similar problems.

To test:

- Bring up an RKE2 cluster with 2 machine pools.
- In the 2nd pool, scale down the pool to 0
- Verify that the pool is still shown as empty
- Verify that you scan scale the pool back to 1

Note: There is a slight issue when scaling that the pool ordering changes briefly - I think this is minor and we can live with this.

Example:

![image](https://user-images.githubusercontent.com/1955897/178284272-bb43a4b5-068f-43ca-8d02-0df0ddc8a37d.png)
